README.md

```diff
@@ -11,6 +11,55 @@ and Structured Streaming for stream processing.
 
 https://spark.apache.org/
 
+## Create a new version 
+
+### Step 1 Add dockerfiles for a new version.
+
+You can see [3.4.0 PR](https://github.com/apache/spark-docker/pull/33) as reference.
+
+- 1.1 Add gpg key to [tools/template.py](https://github.com/apache/spark-docker/blob/master/tools/template.py#L24)
+
+    This gpg key will be used by Dockerfiles (such as [3.4.0](https://github.com/apache/spark-docker/blob/04e85239a8fcc9b3dcfe146bc144ee2b981f8f42/3.4.0/scala2.12-java11-ubuntu/Dockerfile#L41)) to verify the signature of the Apache Spark tarball.
+
+- 1.2 Add image build workflow (such as [3.4.0 yaml](https://github.com/apache/spark-docker/blob/04e85239a8fcc9b3dcfe146bc144ee2b981f8f42/.github/workflows/build_3.4.0.yaml))
+
+    This file will be used by GitHub Actions to build the Docker image when you submit the PR to make sure dockerfiles are correct and pass all tests (build/standalone/kubernetes).
+
+- 1.3 Using `./add-dockerfiles.sh [version]` to add Dockerfiles.
+
+    You will get a new directory with the Dockerfiles for the specified version.
+
+- 1.4 Add version and tag info to versions.json, publish.yml and test.yml.
+
+    This version file will be used by image build workflow (such as [3.4.0](https://github.com/apache/spark-docker/commit/47c357a52625f482b8b0cb831ccb8c9df523affd) reference) and docker official image.
+
+### Step 2. Publish apache/spark Images.
+
+Click [Publish (Java 17 only)](https://github.com/apache/spark-docker/actions/workflows/publish-java17.yaml) (such as 4.x) or [Publish](https://github.com/apache/spark-docker/actions/workflows/publish.yml) (such as 3.x) to publish images.
```

Comment:
there is also a `Publish (Java 11 only)`, shall we mention it?

Reply:
It seems you didn't see the first line? java 17 already mentioned, :)

Reply:
shouldn't mention https://github.com/apache/spark-docker/blob/master/.github/workflows/publish-java11.yml ? Or we should remove this file if it's no longer in use.